### PR TITLE
Fixed issue #291 and updated the dialect for Tableau Connector

### PR DESCRIPTION
**Pull request by guiangumpac**
### Description
- The issue is Timestamp.valueOf(<string>) fails when there are no trailing zeros
- The fix is to parse to LocalDate, then set the time portion to start of day (via atStartOfDay) - and then go to Timestamp
- Added functionality for CAST to string and int
- Added functionality for CEILING and FLOOR
 
### Issues Resolved
https://github.com/opensearch-project/sql/issues/291

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: sql-jdbc/src/main/java/org/opensearch/jdbc/types/TimestampType.java [hunk 1]
```diff
@@ -78,13 +79,17 @@ else if (value.charAt(23) == '+' || value.charAt(23) == '-') {
                 }
             }
 
-            if (calendar == null) {
-                return Timestamp.valueOf(value);
+            final Timestamp ts;
+            if (value.length() < 11) {
```
Comment (joshuali925) [inline sql-jdbc/src/main/java/org/opensearch/jdbc/types/TimestampType.java:83]: question why 11 here? is it to check if value is in yyyy-MM-dd format?

**Review by joshuali925 — DISMISSED** [review]
lgtm, thanks

File: sql-jdbc/src/main/java/org/opensearch/jdbc/types/TimestampType.java (hunk 1 above)
Comment (guiangumpac) [reply sql-jdbc/src/main/java/org/opensearch/jdbc/types/TimestampType.java:83]: Yes. I can add a comment to clarify this.

**Review by joshuali925 — DISMISSED** [review]

**Review by joshuali925 — DISMISSED** [review]

**Review by joshuali925 — APPROVED** [review]
Thanks

**Review by ps48 — APPROVED** [review]

**Pull request merged**